nautobot_circuit_maintenance/jobs/site_search.py

```diff
@@ -144,38 +123,43 @@ def run(self, data=None, commit=None):
             for circuit in circuit_maint.circuits:
-                site_list += get_sites_from_circuit(circuit)
+                site_set.update(get_sites_from_circuit(circuit))
 
+            processed_maintenance_list.append(circuit_maint)
+            overlapping_maintenance = False  # Flag for overlapping maintenance present
             # Check to see if there are any circuit maintenances that are overlapping at some moment in time
-            for site in site_list:
+            for site in site_set:
                 for other_circuit_maint in circuit_maintenance_mapper[site.name]:
                     # Report failures for any time where a circuit will take an outage
-                    if circuit_maint == other_circuit_maint:
+                    # Check to see if the other circuit maintenance has already had overlap checked.
+                    # If this is the same circuit maintenance record, that will also show up in the processed list
+                    # since the list is being added to before the loop.
+                    if other_circuit_maint in processed_maintenance_list:
                         continue
                     if check_for_overlap(circuit_maint, other_circuit_maint):
                         if PLUGIN_SETTINGS.get("overlap_job_exclude_no_impact"):
-                            for impact in circuit_maint.circuitimpact_set:
-                                if impact.impact != "NO-IMPACT":
-                                    self.log_warning(
-                                        obj=site,
-                                        message=f"There is an overlapping maintenance for site: {site.name} on {circuit_maint.start_time}. Other maintenances: {other_circuit_maint}|{circuit_maint}",
-                                    )
-                        else:
-                            self.log_warning(
-                                obj=site,
-                                message=f"There is an overlapping maintenance for site: {site.name} on {circuit_maint.start_time}. Other maintenances: {other_circuit_maint}|{circuit_maint}",
-                            )
-                    else:
-                        # Log success for each time there is a known circuit still available at the site at the same time
-                        self.log_debug(message="Checked maintenance for overlap, no overlap was found.")
+                            for circuit_impact in circuit_maint.circuitimpact_set:
+                                if circuit_impact.impact != "NO-IMPACT":
+                                    break
+
+                                continue
+
+                        self.log_warning(
+                            obj=site,
+                            message=f"There is an overlapping maintenance for site: {site.name} on {circuit_maint.start_time}. Other maintenances: {other_circuit_maint}|{circuit_maint}",
+                        )
+                        overlapping_maintenance = True
```

Comment:
If I understand correctly, `overlapping_maintenance = True` should be updated right after the `if check_for_overlap(circuit_maint, other_circuit_maint):`.

Why? because even in the case of the option for no-impact, and having all the circuit of this type, you are detecting and `overlapping_maintenance`, but not logging